src/test/scala/items/catalogitem/RepositoryTest.scala

```diff
@@ -0,0 +1,159 @@
+        db.findById(id, store).left.value shouldBe CatalogItemNotFound
+      }
+    }
+
+    describe("after being registered and then deleted") {
+      it("should not be present in database") {
+        val db: Repository = repository.getOrElse(fail())
+        val category: ItemCategoryId = ItemCategoryId(35).getOrElse(fail())
+        val store: Store = Store(2).getOrElse(fail())
+        val price: Price = Price(Amount(19.99).getOrElse(fail()), Currency.withName("EUR"))
+        val inPlaceCatalogItem: InPlaceCatalogItem = db.add(category, store, price).getOrElse(fail())
+        db.remove(inPlaceCatalogItem).getOrElse(fail())
+        db.findById(inPlaceCatalogItem.id, store).left.value shouldBe CatalogItemNotFound
+      }
+    }
+
+    describe("after being removed but they were never registered in the first place") {
+      it("should not be allowed") {
+        val db: Repository = repository.getOrElse(fail())
+        val id: CatalogItemId = CatalogItemId(2).getOrElse(fail())
+        val category: ItemCategoryId = ItemCategoryId(35).getOrElse(fail())
+        val store: Store = Store(2).getOrElse(fail())
+        val price: Price = Price(Amount(19.99).getOrElse(fail()), Currency.withName("EUR"))
+        val inPlaceCatalogItem: InPlaceCatalogItem = InPlaceCatalogItem(id, category, store, price)
+        db.remove(inPlaceCatalogItem).left.value shouldBe OperationFailed
+      }
+    }
+
+    describe("after being registered and then their data gets updated") {
+      it("should show the update") {
+        val db: Repository = repository.getOrElse(fail())
+        val category: ItemCategoryId = ItemCategoryId(35).getOrElse(fail())
+        val store: Store = Store(3).getOrElse(fail())
+        val price: Price = Price(Amount(19.99).getOrElse(fail()), Currency.withName("EUR"))
+        val catalogItem: InPlaceCatalogItem = db.add(category, store, price).getOrElse(fail())
+        val newPrice = Price(Amount(14.99).getOrElse(fail()), Currency.withName("USD"))
+        db.update(catalogItem, newPrice).getOrElse(fail())
+        db.findById(catalogItem.id, store).getOrElse(fail()).price shouldBe newPrice
+        db.remove(catalogItem).getOrElse(fail())
+      }
+    }
+
+    describe("when their data gets updated but they were never registered in the first place") {
+      it("should not be allowed") {
+        val db: Repository = repository.getOrElse(fail())
+        val id: CatalogItemId = CatalogItemId(3).getOrElse(fail())
+        val category: ItemCategoryId = ItemCategoryId(35).getOrElse(fail())
+        val store: Store = Store(3).getOrElse(fail())
+        val price: Price = Price(Amount(19.99).getOrElse(fail()), Currency.withName("EUR"))
+        val catalogItem: InPlaceCatalogItem = InPlaceCatalogItem(id, category, store, price)
+        db.update(catalogItem, price).left.value shouldBe OperationFailed
+      }
+    }
+
+    describe("if never registered and then searched for all lifted catalog items") {
```

Comment:
The correct phrase should be "if never registered, while searching for all lifted catalog items"